Common/include/toolboxes/C2DContainer.hpp

```diff
@@ -486,6 +488,70 @@ class C2DContainer :
 };
 
 
+/*!
+ * \class C2DDummyLastView
+ * \brief Helper class, adds dummy trailing dimension to a reference of a
+ *        vector object making it a dummy matrix.
+ * \note The constness of the object is derived from the template type, but
+ *       we allways keep a reference, never a copy of the associated vector.
+ */
+template<class T>
+struct C2DDummyLastView
+{
```

Comment:
I don't really follow what these two structs `C2DDummyLastView` and `C3DDummyMiddleView` are doing, could you elaborate on what they are used for?

Reply:
Ah yes... The gradient function operates on any type that provides operator (iPoint, iVar, iDim) and the limiter function on any type with operator (iPoint, iVar).
Now the gradient of a single scalar (e.g. the "AuxVar") is stored in a 2D container which only gives (iPoint, iDim) so to make it compatible with the function we wrap it to give it a dummy middle dimension. The 2D view does something similar for 1D containers.

This could be explored a bit further to delete some more code, we compute gradients of grid velocities using least squares, those are stored in CGeometry which could be wrapped to again use the same function. This would require small changes elsewhere (the mpi for these is different too) and since those gradients are only used by the continuous adjoint I did not bother...